1.4/Source/VFEEmpire/VFEEmpire/UI/RoyaltyTabWorker_Hierarchy.cs

```diff
@@ -83,82 +88,92 @@ public override void DoMainSection(Rect inRect)
-                Widgets.Label(new Rect(new Vector2(x + 5f, 5f), size), text.Colorize(ColoredText.SubtleGrayColor));
-                Widgets.InfoCardButton(x + 10f + size.x, 5f, title);
-                GUI.DrawTexture(new Rect(x + 5f, 10f + size.y, 75f, 75f), titleExt.GreyIcon);
+                if (shouldDraw)
+                {
+                    GUI.color = ColoredText.SubtleGrayColor;
+                    Widgets.DrawLineVertical(x, 5f, viewRect.height - 15f);
+                    GUI.color = Color.white;
+                    var text = "VFEE.OfThe".Translate(title.LabelCap, Faction.OfEmpire.Name);
+                    var size = Text.CalcSize(text);
+                    Widgets.Label(new Rect(new Vector2(x + 5f, 5f), size), text.Colorize(ColoredText.SubtleGrayColor));
+                    Widgets.InfoCardButton(x + 10f + size.x, 5f, title);
+                    GUI.DrawTexture(new Rect(x + 5f, 10f + size.y, 75f, 75f), titleExt.GreyIcon);
+                }
+
                 titlePos[title] = x;
                 curTitle = title;
             }
 
-            GUI.color = ColoredText.SubtleGrayColor;
-            Widgets.DrawLineHorizontal(x, viewRect.height / 2f, 25f);
-            GUI.color = Color.white;
-
-            x += 25f;
-            pawnPos[pawn] = x;
-
-            if (highlight.Any() && !highlight.Contains(pawn)) GUI.color = Command.LowLightLabelColor;
-
-            var rect = new Rect(x, 0f, 150f, 230f).CenteredOnYIn(viewRect);
-            float nameHeight;
-            rect.height += nameHeight = Text.CalcHeight(pawn.NameFullColored, rect.width);
-            var buttonRect = new Rect(rect.x, rect.yMax + 3f, rect.width, 25f);
-            var honorRect = rect.TakeTopPart(20f);
-            if (curTitle != VFEE_DefOf.Emperor)
-            {
-                GUI.DrawTexture(honorRect.TakeLeftPart(20f), Faction.OfEmpire.def.RoyalFavorIcon);
-                Widgets.Label(honorRect, pawn.TotalFavor().ToString());
-            }
-
-            var titleRect = new Rect(0f, rect.yMax - 20f, Text.CalcSize(title.LabelCap).x + 20f, 20f).CenteredOnXIn(rect.TakeBottomPart(20f));
-            GUI.DrawTexture(titleRect.TakeLeftPart(20f), titleExt.Icon);
-            Widgets.Label(titleRect, title.LabelCap);
-            using (new TextBlock(TextAnchor.MiddleCenter)) Widgets.Label(rect.TakeBottomPart(nameHeight), pawn.NameFullColored);
-            Widgets.DrawWindowBackground(rect);
-            GUI.DrawTexture(rect, PortraitsCache.Get(pawn, rect.size, Rot4.South));
-            if (pawn.Faction is { IsPlayer: true })
+            if (shouldDraw)
             {
-                GUI.color = pawn.Faction.Color;
-                GUI.DrawTexture(rect.BottomPartPixels(75f).RightPartPixels(75f), pawn.Faction.def.FactionIcon);
+                GUI.color = ColoredText.SubtleGrayColor;
+                Widgets.DrawLineHorizontal(x, viewRect.height / 2f, 25f);
                 GUI.color = Color.white;
             }
```

Comment:
Also, question about this part of the code. (Doesn't look too well when comparing to original, oh well...)

Should there be a check for `curTitle != VFEE_DefOf.Freeholder` like before? Currently, it doesn't draw the line going from the pawn with freeholder title (as opposed to the further titles), but it still draws a line going towards them that doesn't connect with anything. For me it feels a bit out of place, but I suppose it could be intended:

![image](https://user-images.githubusercontent.com/36712560/228390612-2a4b5ad4-c1fb-433a-9a0d-cec2ab5a5e90.png)

Reply:
Freeholders aren't supposed to be in the Hierarchy at all, so the check simply avoid drawing a line to the a nonexistent pawn.

![image](https://user-images.githubusercontent.com/11205096/228396011-45aafa9c-6b1b-4b08-afb9-bda3caba3e5c.png)

Reply:
My point is - when drawing player freeholders in hierarchy, it still draws the line to the left of the pawn (as is in my screenshot).